src/Microsoft.AspNet.Mvc.Razor/RazorView.cs

```diff
@@ -233,13 +238,18 @@ private Task RenderPageCoreAsync(IRazorPage page, ViewContext context)
             }
         }
 
-        private IRazorPage GetLayoutPage(ViewContext context, string layoutPath)
+        private IRazorPage GetLayoutPage(ViewContext context, string executingFilePath, string layoutPath)
         {
-            var layoutPageResult = _viewEngine.FindPage(context, layoutPath);
+            var layoutPageResult = _viewEngine.GetPage(executingFilePath, layoutPath, isPartial: true);
```

Comment:
Because `FindPage()` previously hard-coded that value when it was successful. More generally because a layout page can't exist separate from the main page.